src/components/Paginator/Paginator.tsx

```diff
@@ -0,0 +1,340 @@
+
+const {
+	arrayOf,
+	bool,
+	func,
+	number,
+	object,
+	oneOfType,
+	shape,
+	string,
+} = PropTypes;
+
+const { Option } = SingleSelect;
+
+type IPaginatorSingleSelectProps = Partial<ISingleSelectProps>;
+
+type ShowTotalObjects = (count: number) => string;
+
+export interface IPaginatorProps
+	extends StandardProps,
+		React.DetailedHTMLProps<
+			React.HTMLAttributes<HTMLDivElement>,
+			HTMLDivElement
+		> {
+	/** Whether or not to show the page size selector. */
+
+	hasPageSizeSelector: boolean;
+
+	/** Disables the Paginator from being clicked or focused. */
+
+	isDisabled: boolean;
+
+	/** Label when showTotalObjects is true with 1 or fewer objects. */
+
+	objectLabel: string;
+
+	/** Label when showTotalObjects is true with more than 1 objects. */
+
+	objectLabelPlural?: string;
+
+	/** Called when a page is selected. */
+
+	onPageSelect: (
+		pageIndex: number,
+		totalPages: number,
+		{
+			props,
+			event,
+		}: {
+			props: IButtonProps | ITextFieldProps;
+			event: React.MouseEvent | React.FormEvent | React.FocusEvent;
+		}
+	) => void;
+
+	/** Called when a page size is selected. */
+
+	onPageSizeSelect?: (
+		pageSizeIndex: number | null,
+		{
+			props,
+			event,
+		}: {
+			props: ISingleSelectOptionProps | undefined;
+			event: React.MouseEvent | React.KeyboardEvent;
+		}
+	) => void;
+
+	/** 0-indexed currently selected page number. */
+
+	selectedPageIndex: number;
+
+	/** Currently selected page size option index. */
+
+	selectedPageSizeIndex: number;
+
+	/** Show total count of objects. */
+
+	showTotalObjects: boolean | ShowTotalObjects;
+
+	/** Number to display in \`of \${totalPages}\`, calculated from	
+	\`totalPages\` and selected page size by default.  */
+
+	totalPages: number;
+
+	/** Total number of items across all pages. */
+
+	totalCount: number | null;
+
+	/** Array of numbers representing page size options. */
+
+	pageSizeOptions: number[];
+
+	/** Object of SingleSelect props which are passed thru to the underlying SingleSelect component for the page size selector. */
+
+	SingleSelect: IPaginatorSingleSelectProps;
+
+	/** Object of TextField props which are passed thru to the underlying TextField component. */
+
+	//TextField: TextField.defaultProps;
+	TextField: ITextFieldProps;
+}
+
+export interface IPaginatorState {
+	pageIndex: number;
+	totalPages: number;
+	totalCount: number;
+	selectedPageIndex: number;
+	selectedPageSizeIndex: number;
+	pageSizeOptions: number[];
+	SingleSelect: ISingleSelectState;
+}
+
+const defaultProps = {
+	hasPageSizeSelector: false,
+	isDisabled: false,
+	objectLabel: 'Object',
+	onPageSelect: _.noop,
+	selectedPageIndex: 0,
+	selectedPageSizeIndex: 0,
+	showTotalObjects: false,
+	totalCount: null,
+	totalPages: undefined,
+	pageSizeOptions: [10, 50, 100],
+	SingleSelect: {
+		...SingleSelect.defaultProps,
+		selectedIndex: 0,
+	},
+	TextField: TextField.defaultProps,
+};
+
+class Paginator extends React.Component<IPaginatorProps, IPaginatorState> {
```

Comment:
FWIW I think this component could have been converted to a functional component since it doesn't appear to be using any lifecycle methods.